### ISSUE TITLE
isfile (Python says no, pyfakefs says yes)

### ISSUE THREAD

**Issue by agroce** [jmcgeheeiv/pyfakefs#374]
```
import shutil
import pyfakefs.fake_filesystem
import os

ROOT_PATH = "/Volumes/ramdisk/newtest"

fs = pyfakefs.fake_filesystem.FakeFilesystem()
fs.CreateDirectory('/Volumes')
fs.CreateDirectory('/Volumes/ramdisk')
fs.CreateDirectory('/Volumes/ramdisk/newtest')
os0 = pyfakefs.fake_filesystem.FakeOsModule(fs)
opener0 = pyfakefs.fake_filesystem.FakeFileOpen(fs)

if True:
    os0 = os
    shutil.rmtree(ROOT_PATH)
    os.mkdir(ROOT_PATH)
    opener0 = open

component0 = "alpha" 
path0 = ROOT_PATH 
path0 += os.path.sep + component0 
component0 = "" 
file0 = opener0(path0,'w') 
path0 += os.path.sep + component0 
result = os0.path.isfile(path0)
print result
```

**Comment by mrbean-bremen** [comment]
For reference: returns False under Linux and Windows, False under MacOS.

**Comment by mrbean-bremen** [comment]
I don't understand this stuff with file paths ending with separators - it seems quite random to me. Sometimes MacOS behavior is the opposite of Windows and Linux, sometimes it's Windows that behaves different, and I don't see a pattern here at all. `</rant>`
I'm giving up for now, maybe will have another go at the weekend.

**Comment by mrbean-bremen** [comment]
Just to let you know - I'm just mostly working on another project, so it may be a few days before I get back to pyfakefs.

**Comment by mrbean-bremen** [comment]
Fixed by #382
